### ISSUE TITLE
Add support for specifying the `loss` used in random forests and AdaBoost model

### ISSUE THREAD

**Issue by ablaom** [JuliaAI/DecisionTree.jl#217]
As far as I can tell, the `loss` parameter is only exposed for single trees. I think this would be pretty easy to add to the ensemble models. 

Issue raised at #211.

**Comment by fipelle** [comment]
Also, it seems that `loss` is only available for classification trees - not regression trees. Have I missed something in the code?

**Comment by ablaom** [comment]
> multi-target problems (the current implementation for regression trees does not allow for `features` that are not `Float64` - i.e., single targets).

Do you mean `features` here or, rather, `labels` (aka target)?

**Comment by fipelle** [comment]
`labels` as in https://scikit-learn.org/stable/auto_examples/ensemble/plot_random_forest_regression_multioutput.html

**Comment by ablaom** [comment]
Right. Your interesting question is orthogonal to initial post, so addressing it [here](https://github.com/JuliaAI/DecisionTree.jl/issues/30).
